### PR TITLE
AMS-184: add context & decorator method for EE

### PR DESCRIPTION
**Pull request by grena**
**Description (for Contributor and Core Developer)**

This PR adds some method for behat tests.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: features/Pim/Behat/Decorator/Grid/Filter/ChoiceDecorator.php [hunk 1]
```diff
@@ -24,4 +24,38 @@ public function filter($operator, $value)
...
+            return $this->getBody()->findAll('css', '.select-filter-widget.dropdown-menu');
+        }, 'Could not find any multiselect widget');
+
+        $visibleWidgets = array_filter($multiSelectWidgets, function ($widget) {
+            return $widget->isVisible();
+        });
+
+        if (empty($visibleWidgets)) {
+            throw new \Exception('Could not find the multiselect widget');
+        }
```
Comment (pierallard) [inline features/Pim/Behat/Decorator/Grid/Filter/ChoiceDecorator.php]: Can you do all this into a spin plz ?

File: features/Context/DataGridContext.php [hunk 2]
```diff
@@ -746,6 +746,47 @@ public function iShouldSeeOptionInFilter($optionNames, $filterName)
...
+    }
+
+    /**
+     *
+     * @param $option
+     * @param $filterName
+     *
+     * @Given /^I should see the available option "([^"]*)" in the filter "([^"]*)"$/
+     */
+    public function iShouldSeeTheAvailableOptionInTheFilter($option, $filterName)
```
Comment (pierallard) [inline features/Context/DataGridContext.php]: why not merge these 2 methods with a /not/ regepx ?

File: features/Pim/Behat/Decorator/Grid/Filter/ChoiceDecorator.php (hunk 1 above)
Comment (grena) [inline features/Pim/Behat/Decorator/Grid/Filter/ChoiceDecorator.php]: I don't need a spin for that, I already spinned above, no I just look for visible objects

File: features/Context/DataGridContext.php (hunk 2 above)
Comment (grena) [inline features/Context/DataGridContext.php]: it was to keep the methods simple, but I can merge yes :)

**Pull request merged**